### DISCUSSION TITLE
Release Archive Checksums

### DISCUSSION THREAD

**Discussion by s0undt3ch** [community/community#45091] [Actions]
### Select Topic Area

Product Feedback

### Body

Currently, whenever a new GitHub actions runner is released, the archives Checksums is shown in the release notes.

When we try to automate the upgrade of the GitHub runner for self hosted runners, there's no easy way to confirm that the archive we download matches the release checksum.

The same way you upload the release archives when you make a release, you could also upload a file with the archive checksum, this will allow an automated procedure to update to latest runner while confirming the Checksums match.

Currently, our automation process just downloads the latest release and uses that.

The counter argument could be, one should bump the runner version manually and confirm the Checksums, but we already put a lot of trust on GitHub to run our CI pipelines, and the same level of trust can be applied to the releases you do for your runners.
You already mention in the release notes how to install the versions, so that's almost suggesting that automation is welcomed.
A step further would be to also confirm that the checksum of what was downloaded matches, preferably, in a way that can also be automated.

**Comment by airtower-luna** [comment]
> When we try to automate the upgrade of the GitHub runner for self hosted runners, there's no easy way to confirm that the archive we download matches the release checksum.

You can just download the release notes [via the API](https://docs.github.com/en/rest/releases/releases?apiVersion=2022-11-28#get-the-latest-release), the notes are in the `body` element of the response. E.g. (replace "latest" with the tag if you don't want the latest release):

```
gh api /repos/actions/runner/releases/latest --template '{{.body}}'
```

Extracting the checksum for the package you use should be pretty simple with a regular expression. :slightly_smiling_face:

**Reply by s0undt3ch** [reply]
> Extracting the checksum for the package you use should be pretty simple with a regular expression. 🙂

True, but downloading two files from two Urls is much simpler, and an apparent ~best~ common practice.

**Reply by s0undt3ch** [reply]
Nonetheless, a pretty good workaround while something less involved is available.
